src/index_update_db.js

```diff
@@ -2,60 +2,177 @@
+        .catch(errRole => {
+          console.log(errRole);
+          reject(errRole);
+        })
+      );
+  });
+}
+
+function createIamRoles(dbIamRoles, accData, mcawsDbObj) {
+  return new Promise((resolve, reject) => {
+
+    for (let idx = 0; idx < dbIamRoles.length; idx++) {
+      dbIamRoles[idx].account = accData.dataValues.id;
+    }
+    Promise.all(dbIamRoles.map(createIamRole)).then(function(results){
+      console.log("Roles creation completed");
+      resolve(true);
+    })
+  });
+}
 
 exports.handler = function(event, context, callback) {
   const dbIamRoles = event.dbIamRoles;
   const dbAwsAccount = event.dbAwsAccount;
+  const fullAdminRoleName = 'FullAdmin';
+  const drAdminRole = event.drAdminRole;
   const encryptedBuf = new Buffer(process.env.DB_PASSWORD, 'base64');
   const cipherText = { CiphertextBlob: encryptedBuf };
   const kms = new AWS.KMS({ region: process.env.KMS_REGION });
+  let accountDetails = null;
   kms.decrypt(cipherText, (err, passwd) => {
     if (err) {
       console.log('Decrypt error:', err);
       callback(err);
       return;
     }
-    const mcawsDbObj = new McawsModels(
+    mcawsDbObj = new McawsModels(
       process.env.DB_USERNAME,
       passwd.Plaintext.toString('ascii'),
       process.env.DB_HOST,
       'msaws'
     );
-    mcawsDbObj.AwsAccount(accResp => {
-      accResp.sync().then(() =>
-        accResp.findOne({
+
+    mcawsDbObj.AwsAccount(accountResp => {
+      accountResp.sync().then(() =>
+        accountResp.findOne({
              where: dbAwsAccount
         }).then(accountData =>{
              if(accountData) {
                console.log("Account entry is already there.");
                 return accountData;
              }
-             else return accResp.create(dbAwsAccount);
+             else return accountResp.create(dbAwsAccount);
          })
           .then(accData => {
-            for (let idx = 0; idx < dbIamRoles.length; idx++) {
-              dbIamRoles[idx].account = accData.dataValues.id;
-              mcawsDbObj.AwsIamRole(iamResp =>
-                iamResp.findOne({
-                   where: {arn: dbIamRoles[idx].arn}
-                })
-                .then(resultData =>{
-                  if(resultData) {
-                     console.log("IAM Role entry is already there.");
-                     return resultData;
-                  }
-                  else return iamResp.create(dbIamRoles[idx]);
-                })
-                .then(roleData => {
-                  console.log('role updation Done :)');
-                  console.log(roleData);
-                })
-                .catch(errRole => console.log(errRole))
-                .finally(() => mcawsDbObj.CloseConnection())
-              );
-            }
+            accountDetails = JSON.parse(JSON.stringify(accData));
+            createIamRoles(dbIamRoles, accData, mcawsDbObj)
+            .then(() => {
+              addDradminRole(dbAwsAccount, drAdminRole,
```

Comment:
the function is adding the dradmin to both account and IAM role, could that be added to the function as well addDrAdminRoleToAccountAndIAMRole?